### ISSUE TITLE
Use lifecycle type sidecar on kuma-sidecar container

### ISSUE THREAD

**Issue by jakubdyszkiewicz** [kumahq/kuma#2203]
### Summary

add `lifecycle.type=Sidecar`
```
  - name: kuma-sidecar
    image: docker.io/kumahq/kuma-dp:1.2.0
    lifecycle:
      type: Sidecar
```
conditionally when we are injecting Kuma DP on Kubernetes 1.18+

This way Kuma DP will start as a first container.

**Comment by jpeach** [comment]
See https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/753-sidecar-containers

**Comment by jakubdyszkiewicz** [comment]
I was kind of misled that this is a feature available in 1.18 but it was not released yet and won't be released in this form...

**Comment by jakubdyszkiewicz** [comment]
Closing for now. Will reopen once Kubernetes has this feature.
